python/cuspatial/cuspatial/core/geodataframe.py

```diff
@@ -178,6 +178,55 @@ def _slice(self: T, arg: slice) -> T:
         )
         return self.__class__(result)
 
+    def _recombine_geo_and_cudf_dfs(self, geo_columns, data_columns):
+        """
+        Combine a GeoDataFrame of only geometry columns with a DataFrame
+        of non-geometry columns in the same order as the columns in `self`
+        """
+        column_names = pd.Series(self._data.names)
+        geocolumn_mask = pd.Series(
+            [isinstance(self[col], GeoSeries) for col in self._data.names]
+        )
+        result = GeoDataFrame(
+            {
+                name: (geo_columns[name] if mask else data_columns[name])
+                for name, mask in zip(
+                    column_names.values, geocolumn_mask.values
+                )
+            },
+        )
+        return result
```

Comment:
I'm not sure if this helper is needed if all you need is to make a dataframe from the dict returned from `_recombine_columns`.

Reply:
That's a good idea.

Reply:
Done.